<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlantScannerClass</class>
 <widget class="QMainWindow" name="PlantScannerClass">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>252</width>
    <height>547</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>植物扫描仪_SJTU</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>Resources/SJTU.jpg</normaloff>Resources/SJTU.jpg</iconset>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="LaserOpenButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>打开</string>
    </property>
   </widget>
   <widget class="QPushButton" name="LaserCloseButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>280</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>关闭</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>激光发射器</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>顶部丝杠</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonUp1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>上升</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonDown1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>下降</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonStop1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>停止</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>150</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>激光丝杠</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonUp2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>170</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>上升</string>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonDown2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>200</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>下降</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonStop2">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>230</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>停止</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>远柜门丝杠</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonUp3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>上升</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonDown3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>60</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>下降</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonReset3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>复位</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonStop3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>停止</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>近柜门丝杠</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonUp4">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>30</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>上升</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonDown4">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>60</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>下降</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonReset4">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>90</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>复位</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OnButtonStop4">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>120</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>停止</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>串口选择</string>
    </property>
   </widget>
   <widget class="QComboBox" name="port_number">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="port_baudate">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>71</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>360</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>波特率</string>
    </property>
   </widget>
   <widget class="QPushButton" name="OpenPortButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>330</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Open</string>
    </property>
   </widget>
   <widget class="QPushButton" name="CloseButton">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>380</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Quit</string>
    </property>
   </widget>
   <widget class="QDateTimeEdit" name="dateTimeEdit">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>470</y>
      <width>131</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>LaserOpenButton</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_LaserOpenButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>421</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LaserCloseButton</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_LaserCloseButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>381</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonUp1</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonUp1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>166</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonDown1</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonDown1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>41</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonStop1</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonStop1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>161</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonUp2</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonUp2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonDown2</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonDown2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>269</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonStop2</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonStop2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>272</x>
     <y>319</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonUp3</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonUp3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonDown3</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonDown3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>122</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonReset3</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonReset3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>161</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonStop3</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonStop3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>27</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonUp4</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonUp4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>335</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonDown4</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonDown4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>318</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonReset4</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonReset4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>116</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OnButtonStop4</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_ClickedButtonStop4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>130</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>OpenPortButton</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_OpenPortButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>171</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>514</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CloseButton</sender>
   <signal>clicked()</signal>
   <receiver>PlantScannerClass</receiver>
   <slot>on_CloseButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>518</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_LaserOpenButton()</slot>
  <slot>on_LaserCloseButton()</slot>
  <slot>on_ClickedButtonUp1()</slot>
  <slot>on_ClickedButtonDown1()</slot>
  <slot>on_ClickedButtonStop1()</slot>
  <slot>on_ClickedButtonUp2()</slot>
  <slot>on_ClickedButtonDown2()</slot>
  <slot>on_ClickedButtonStop2()</slot>
  <slot>on_ClickedButtonUp3()</slot>
  <slot>on_ClickedButtonDown3()</slot>
  <slot>on_ClickedButtonReset3()</slot>
  <slot>on_ClickedButtonStop3()</slot>
  <slot>on_ClickedButtonUp4()</slot>
  <slot>on_ClickedButtonDown4()</slot>
  <slot>on_ClickedButtonReset4()</slot>
  <slot>on_ClickedButtonStop4()</slot>
  <slot>on_OpenPortButton()</slot>
  <slot>on_CloseButton()</slot>
  <slot>on_GetPointCloudButton()</slot>
  <slot>on_PointCloudViewButton1()</slot>
  <slot>on_PointCloudViewButton2()</slot>
  <slot>on_PointCloudViewButton3()</slot>
  <slot>on_PointCloudViewButton4()</slot>
  <slot>on_PointCloudViewButton5()</slot>
  <slot>on_PointCloudViewButton6()</slot>
  <slot>on_ImageButton()</slot>
  <slot>on_ImageCapture()</slot>
 </slots>
</ui>
